clippy_lints/src/doc.rs

```diff
@@ -485,6 +487,9 @@ fn check_doc<'a, Events: Iterator<Item = (pulldown_cmark::Event<'a>, Range<usize
     let mut in_heading = false;
     let mut is_rust = false;
     let mut edition = None;
+    let mut ticks_unbalanced = false;
+    let mut text_to_check: Vec<(CowStr<'_>, Span)> = Vec::new();
+    let mut paragraph_span = spans[0].1; // First line
```

Comment:
Can you be sure that `spans` isn't empty?

Reply:
I don't think that should ever panic. I just did some tests where a function has no doc comment, or has an empty doc comment, and they were all fine. The only time that spans would be empty is if there isn't a doc comment, and if that's the case the `check_doc` function is never called:

https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/doc.rs#L435-L441

Reply:
That's good! Can you then add a comment or, even better, replace this with `spans.get(0).expect("<explanation>")`, please?